docs/changelog/3_x.rst

```diff
@@ -0,0 +1,517 @@
+
+The EdgeDB 3.0 beta was released on May 2nd, 2023. We would like to thank our
+community for reporting issues and contributing fixes. You are awesome! ❤️
+
+To play with the new features, install the CLI using
+<https://www.edgedb.com/install>`_, upgrade to the testing channel using
+``edgedb cli upgrade --to-testing``, and initialize a new project. 
+
+.. code-block:: bash
+
+  $ edgedb project init --server-version=testing
+
+
+Upgrading
+=========
+
+**Local instances**
+
+To upgrade a local project, first upgrade your CLI using ``edgedb cli upgrade
+--to-testing``. Then run the following command inside the project directory.
+
+.. code-block:: bash
+
+  $ edgedb project upgrade --to-testing
+
+Alternatively, specify an instance name if you aren't using a project.
+
+.. code-block:: bash
+
+  $ edgedb project upgrade --to-testing -I my_instance
+
+**Hosted instances**
+
+To upgrade a remote (hosted) instance, we recommend the following
+dump-and-restore process.
+
+1. Spin up an empty 3.0 instance. You can use one of our :ref:`deployment
+guides <ref_guide_deployment>`, but you will need to modify some of the
+commands to use our testing channel and the beta release.
+
+    Under Debian/Ubuntu, when adding the EdgeDB package repository, use this
+    command instead:
+
+    .. code-block:: bash
+
+        echo deb [signed-by=/usr/local/share/keyrings/edgedb-keyring.gpg] \
+          https://packages.edgedb.com/apt \
+          $(grep "VERSION_CODENAME=" /etc/os-release | cut -d= -f2) testing \
+          | sudo tee /etc/apt/sources.list.d/edgedb.list
+
+    Use this command for installation under Debian/Ubuntu:
+
+    .. code-block:: bash
+
+        sudo apt-get update && sudo apt-get install edgedb-3
+
+    Under CentOS/RHEL, use this installation command:
+
+    .. code-block:: bash
+
+        sudo yum install edgedb-3
+
+    In any required ``systemctl`` commands, replace ``edgedb-server-2`` with
+    ``edgedb-server-3``.
+
+    Under any Docker setups, supply the ``3.0-beta.1`` tag.
+
+2. Take your application offline, then dump your v2.x database with the CLI
+
+   .. code-block:: bash
+
+     $ edgedb dump --dsn <old dsn> --all my_database.dump/
+
+   This will dump the schema and contents of your current database to a file
+   on your local disk called ``my_database.dump``. The file name isn't
+   important.
+
+3. Restore the empty v3.x instance from the dump
+
+   .. code-block:: bash
+
+     $ edgedb restore --all my_database.dump/ --dsn <new dsn>
+
+   Once the restore is complete, update your application to connect to the new
+   instance.
+
+This process will involve some downtime, specifically during steps 2 and 3.
+
+
+Client libraries
+----------------
+
+We've released new versions of our JavaScript and Python client libraries that
+support all 3.0 features and implement the updated protocol. These versions
+are backwards compatible with v1.x and v2.x instances, so we encourage all users to
+upgrade.
```

Comment:
Let's mention that many client libraries gained codegen capabilities, since the release of 2.0

Reply:
I'd like to link to the documentation for code generation in the libraries that gained it too. I see documentation on code generation for TS, Python, and Dart. Did TS already have code generation as of 2.0, or is it also new for this version? Any others I need to mention besides those?